src/front-end/typescript/lib/pages/proposal/code-with-us/edit/tab/proposal.tsx

```diff
@@ -705,6 +713,47 @@ const update: component_.page.Update<State, InnerMsg, Route> = ({
         ]
       ];
     }
+    case "checkStatus": {
+      const proposal = state.proposal;
+      return [
+        state,
+        proposal && state.isEditing
+          ? [
+              api.proposals.cwu.readOne<Msg>(proposal.opportunity.id)(
+                proposal.id,
+                (result) => adt("onCheckStatusResponse", result)
+              )
+            ]
+          : []
+      ];
+    }
+    case "onCheckStatusResponse": {
+      const incomingProposal = api.getValidValue(msg.value, null);
+      if (!incomingProposal || !state.proposal) {
+        return [state, []];
+      }
```

Comment:
If a request randomly fails, you may not want to stop polling altogether. It might be better to keep polling even if you encounter a failure.